### ISSUE TITLE
kmsg flooding can starve out /dev/log processing

### ISSUE THREAD

**Issue by syyhao1994** [systemd/systemd#18065]
**systemd version the issue has been seen with**
 > 246

**Used distribution**
 > fedora

**Expected behaviour you didn't see**
 > logger "xxx" should very quick

**Unexpected behaviour you saw**
 > logger is stucked for a long time

**Steps to reproduce the problem**
 > When there are a lot of logs that systemd-journald needed to process, maybe logs are from stdout, syslog, native socket, dmesg, or audit log. I got that the journald may process kernel messages at first priority, and other logs are the same priority.
This cause a problem is if the /dev/log don't processed timely and as a result it is full. We can't logger log anymore and we can't even use ssh to login as ssh may produce logs that can't be sent to /dev/log.
So i think journald may process /dev/log at fisrt priority, or just behind kmesg but have higher priority than other logs at least.

**Comment by poettering** [comment]
WEll, if the system is overloaded, it is overloaded. We need to do the work we need to do. 

Do I understand properly, that you generate an extreme amount of kernel log messages, and the problem is that /dev/log handling gets starved out then? And you'd prefer something more fair, where /dev/log gets processed every now and then too?

**Comment by syyhao1994** [comment]
> WEll, if the system is overloaded, it is overloaded. We need to do the work we need to do.
> 
> Do I understand properly, that you generate an extreme amount of kernel log messages, and the problem is that /dev/log handling gets starved out then? And you'd prefer something more fair, where /dev/log gets processed every now and then too?

Yes, as well as other logs that have the same priority as /dev/log, maybe a more comprehensive behavior should be more fair to process every kind of messages to avoid of starving.
